pkg/apis/network/v1/egressrouter_types.go

```diff
@@ -0,0 +1,228 @@
+
+	// Name of the master interface. Need not be specified if it can be inferred from the IP address.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Optional
+	// +optional
+	Master string `json:"master,omitempty"`
+}
+
+// IPConfig contains the configuration for the IP arguments
+// +kubebuilder:validation:Required
+type IPConfig struct {
+	// List of IP addresses to configure on the pod's secondary interface.
+	// +kubebuilder:validation:Required
+	Addresses []EgressRouterAddress `json:"addresses"`
+
+	// List of destinations that the pod is allowed to connect to via this interface. If not provided, the pod can connect to any destination.
+	// Examples:
+	// {1} ["<dst-ip-addr/mask>"] (["203.0.113.25/30"])
+	// {2} ["<src-port> <protocol> <dst-ip-addr/mask>", "<src-port> <protocol> <dst-ip-addr/mask> <dst-port>"] (["80 tcp 203.0.113.25/30", "8080 tcp 203.0.113.26 80", "8443 tcp 203.0.113.26/30 443"])
+	// +kubebuilder:validation:Optional
+	// +optional
+	AllowedDestinations []string `json:"allowedDestinations,omitempty"`
+}
+
+// EgressRouterAddress contains a pair of IP CIDR and gateway to be configured on the router's interface
+// +kubebuilder:validation:Required
+type EgressRouterAddress struct {
+	// IP is the address to configure on the router's interface. Can be IPv4 or IPv6.
+	// +kubebuilder:validation:Required
+	IP CIDR `json:"ip"`
+	// IP address of the next-hop gateway, if it cannot be automatically determined. Can be IPv4 or IPv6.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Gateway string `json:"gateway,omitempty"`
+}
+
+// CIDR represents an IPv4 or IPV6 CIDR block.
+type CIDR string
+
+// EgressRouterStatusConditionType is an aspect of the router's state.
+type EgressRouterStatusConditionType string
+
+const (
+	// EgressRouterAvailable indicates that the EgressRouter (the associated pod, service, NAD), is functional and available in the cluster.
+	EgressRouterAvailable EgressRouterStatusConditionType = "Available"
+
+	// EgressRouterProgressing indicates that the router is actively rolling out new code,
+	// propagating config changes, or otherwise moving from one steady state to
+	// another.
+	EgressRouterProgressing EgressRouterStatusConditionType = "Progressing"
+
+	// EgressRouterDegraded indicates that the router's current state does not match its
+	// desired state over a period of time resulting in a lower quality of service.
+	EgressRouterDegraded EgressRouterStatusConditionType = "Degraded"
+
+	// EgressRouterUpgradeable indicates whether the router is in a state that is safe to upgrade. When status is `False`
+	// administrators should not upgrade their cluster and the message field should contain a human readable description
+	// of what the administrator should do to allow the operator to successfully update.  A missing condition, True,
+	// and Unknown are all treated by the CNO as allowing an upgrade.
+	EgressRouterUpgradeable EgressRouterStatusConditionType = "Upgradeable"
```

Comment:
"upgradeable" mostly only makes sense at the level of operators. I don't think we need an upgradeable condition here. (If we have use for it later we can add it later...)

Reply:
ack will change this.